File: VB/vbnet/vbnet/Get_Data/Single_Transaction_Query.vb
' *
' * Bluepay VB.NET Sample code.
' *
' * This code sample runs a report that grabs a single transaction
' * from the BluePay gateway based on certain criteria.
' * See comments below on the details of the report.
' * If using TEST mode, only TEST transactions will be returned.
' *

Imports System
Imports vbnet.BPVB

Namespace GetData

    Public Class SingleTransactionQuery

        Public Shared Sub run()

            Dim accountID As String = "Merchant's Account ID Here"
            Dim secretKey As String = "Merchant's Secret Key Here"
            Dim mode As String = "TEST" 

            Dim query As BluePay = New BluePay(
                accountID,
                secretKey,
                mode
            )

            ' Transaction ID: "100233712957"
            ' Report Stard Date: YYYY-MM-DD
            ' Report End Date: YYYY-MM-DD
            ' Do not include errored transactions? Yes
            query.getSingleTransactionQuery(
              transactionID:= "Transaction ID here", 
              reportStartDate:= "2015-01-01", 
              reportEndDate:= "2015-05-30", 
              excludeErrors:= "1" 
            )

            query.process()

            Console.Write(query.response)
    
        End Sub
    End Class
End Namespace

File: C++/bluepay-cpp/Get_Data/Single_Transaction_Query.cpp
//
// BluePay C++ Sample code.
//
// This code sample runs a report that grabs a single transaction
// from the BluePay gateway based on certain criteria.
// See comments below on the details of the report.
// If using TEST mode, only TEST transactions will be returned.

#include "Single_Transaction_Query.h"
#include "BluePay.h"
using namespace std;

void singleTransactionQuery(){
    
    string accountId = "Merchant's Account ID Here";
    string secretKey = "Merchant's Secret Key Here";
    string mode = "TEST";
    string transactionID = "Transaction ID here";

	BluePay report(
		accountId,
		secretKey,
		mode
	);

	report.getSingleTransQuery(
		transactionID, // ID of previous transaction
		"2015-01-01", // Search Date Start: YYYY-MM-DD
		"2015-05-30", // Search Date End: YYYY-MM-DD
		"1" // Do not include errored transactions in search? Yes
	);

    // Makes the API Request with Blue
	report.process();

    // Reads the responses from BluePay if transaction was approved
	cout << report.getResponse();
}

